src/Components/ElementsPanel/Internal/VisualRuleElementsList.tsx

```diff
@@ -0,0 +1,282 @@
+    );
+};
+
+function getListItems(
+    panelItems: Array<IViewerElementsPanelItem>,
+    isModal: boolean,
+    openCalloutRowId: string,
+    onItemClick: ElementsPanelCallback,
+    onItemHover: ElementsPanelCallback,
+    onItemBlur: ElementsPanelCallback,
+    onItemHoverExtended: ViewerElementsPanelCallback,
+    onItemBlurExtended: ViewerElementsPanelCallback
+): Array<ICardboardGroupedListItem<ITwinToObjectMapping | IVisual>> {
+    const sortedPanelItems = sortPanelItemsForDisplay(panelItems);
+    const buttonStyles = getElementsPanelButtonSyles();
+    const listItems: Array<
+        ICardboardGroupedListItem<ITwinToObjectMapping | IVisual>
+    > = [];
+
+    sortedPanelItems.forEach((panelItem) => {
+        const element = panelItem.element;
+        const elementColorings: Array<VisualColorings> = [];
+        const badges: Array<{
+            behavior: IBehavior;
+            visual: IValueRangeVisual;
+            visualRule: IExpressionRangeVisual;
+            visualRuleDisplayTitle: string;
+        }> = [];
+        let visualRules: IExpressionRangeVisual[] = [];
+
+        panelItem.behaviors.forEach((b) => {
+            // Add all visual rules to an array
+            visualRules = b.visuals.filter(ViewerConfigUtility.isVisualRule);
+
+            // Separate into their respective arrays
+            visualRules.forEach((vr) => {
+                vr.valueRanges.forEach((condition) => {
+                    // Check if visual will be shown and then determine if it is a badge or element coloring
+                    if (
+                        shouldShowVisual(
+                            vr.valueRangeType,
+                            panelItem.twins,
+                            vr.valueExpression,
+                            condition.values
+                        )
+                    ) {
+                        if (hasBadge(condition)) {
+                            badges.push({
+                                behavior: b,
+                                visual: condition.visual,
+                                visualRuleDisplayTitle: parseLinkedTwinExpression(
+                                    wrapTextInTemplateString(
+                                        condition.visual.labelExpression
+                                    ),
+                                    panelItem.twins
+                                ),
+                                visualRule: vr
+                            });
+                        } else {
+                            elementColorings.push({
+                                color: condition.visual.color,
+                                label: condition.visual.labelExpression
+                            });
+                        }
+                    }
+                });
+            });
+        });
+
+        //sort the alert within its own group by name
+        badges.sort((a, b) =>
+            a.visualRuleDisplayTitle.localeCompare(
```

Comment:
I think there is a sort method in utils, lets use it: `sortAscendingOrDescending`